services/core-api/tests/projects/project_summaries/resources/test_project_summary_resource.py

```diff
@@ -83,131 +92,338 @@ def test_update_project_summary_assign_project_lead(test_client, db_session, aut
+            },
+            {
+                "address_line_1": "123 fake street",
+                "city": "Vancouver",
+                "sub_division_code": "BC",
+                "post_code": "V5S4W2",
+                "address_type_code": "CAN"
+            }
+        ],
+    }
+    
+    # Agent data
+    data['is_agent'] = True
+    data['agent'] = {
+        "party_type_code": "PER",
+        "phone_no": party.phone_no,
+        "email": "test@gov.bc.ca",
+        "party_name": "test name",
+        "first_name": "test name",
+        "address": {
+            "suite_no": "",
+            "address_line_1": None,
+            "city": "Vancouver",
+            "sub_division_code": "BC",
+            "post_code": None,
+            "address_type_code": "CAN"
+        },
+    }
+
+    # Facility operator data
+    data['facility_coords_source'] = 'GPS'
+    data['facility_desc'] = 'description'
+    data['facility_latitude'] = '50.0000000'
+    data['facility_longitude'] = '-115.0000000'
+    data['facility_type'] = 'test type'
+    data['zoning'] = True
+    data['facility_operator'] = {
+        "party_type_code": "PER",
+        "phone_no": "123-123-1234",
+        "email": "test@govb.bc.ca",
+        "party_name": "test name",
+        "first_name": "test",
+        "address": {
+            "suite_no": "123",
+            "address_line_1": "some address",
+            "city": "vancouver",
+            "sub_division_code": "BC",
+            "post_code": "V5S4W2",
+            "address_type_code": "CAN"
+        },
+    }
+
+    # Legal land data
+    data['is_legal_land_owner'] = False
+    data['is_crown_land_federal_or_provincial'] = None
+    data['is_landowner_aware_of_discharge_application'] = True
+    data['has_landowner_received_copy_of_application'] = False
+    data['legal_land_owner_name'] = 'some name'
+    data['legal_land_owner_contact_number'] = '123-123-1234'
+    data['legal_land_owner_email_address'] = 'test@gov.bc.ca'
+
+    # Declaration data
+    data['confirmation_of_submission'] = None
+    data['ams_terms_agreed'] = False
 
     put_resp = test_client.put(
         f'/projects/{project.project_guid}/project-summaries/{project_summary.project_summary_guid}',
         headers=auth_headers['full_auth_header'],
         json=data)
     put_data = json.loads(put_resp.data.decode())
 
+    error_message = """400 Bad Request: {'surface_level_data': [], 
+    'basic_info': [], 
+    'authorizations': [], 
+    'project_contacts': [], 
+    'applicant_info': [], 
+    'agent': ['{"address": ["no definitions validate", {"anyof definition 0": ["must be of list type"], "anyof definition 1": [{"address_line_1": ["null value not allowed"]}]}]}'], 
+    'facility': [], 
+    'legal_land': ['{"is_crown_land_federal_or_provincial": ["null value not allowed"]}'], 
+    'declaration': ['{"ams_terms_agreed": ["unallowed value False"], "confirmation_of_submission": ["null value not allowed"]}']}"""
+    
```

Comment:
looks like we probably do need a fancy error handler on the FE!

Reply:
Yes! Not very user friendly at all at the moment...